src/org/zaproxy/zap/extension/ascanrulesAlpha/HttpoxyScanner.java

```diff
@@ -0,0 +1,189 @@
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.core.scanner.Category;
+import org.parosproxy.paros.network.HttpMessage;
+
+public class HttpoxyScanner extends AbstractAppPlugin {
+
+	/**
+	 * Prefix for internationalised messages used by this rule
+	 */
+	private static final String MESSAGE_PREFIX = "ascanalpha.httpoxy.";
+	private static final int PLUGIN_ID = 10107;
+
+	private static final Logger log = Logger.getLogger(HttpoxyScanner.class);
+
+	@Override
+	public int getCategory() {
+		return Category.SERVER;
+	}
+
+	@Override
+	public String[] getDependency() {
+		return null;
+	}
+
+	@Override
+	public String getDescription() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "desc");
+	}
+
+	@Override
+	public int getId() {
+		return PLUGIN_ID;
+	}
+
+	@Override
+	public int getRisk() {
+		return Alert.RISK_HIGH;
+	}
+
+	@Override
+	public String getName() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "name");
+	}
+
+	@Override
+	public String getReference() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "refs");
+	}
+
+	@Override
+	public String getSolution() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "soln");
+	}
+
+	@Override
+	public void init() {
+	}
+
+	@Override
+	public void scan() {
+		// Set up a listener using all interfaces on another port
+		ProxyServer proxy;
+		proxy = new ProxyServer();
+		HttpoxyListener listener = new HttpoxyListener();
+		proxy.addOverrideMessageProxyListener(listener);
+		int proxyPort = proxy.startServer(null, 0, true);
+
+		try {
+			Enumeration<NetworkInterface> allInterfaces = NetworkInterface.getNetworkInterfaces();
+			
+			// Try all of the available interfaces as we dont know which one(s) 
+			// will be available to the target server
+			while (allInterfaces.hasMoreElements()) {
+				NetworkInterface ni = allInterfaces.nextElement();
+				Enumeration<InetAddress> inetAddrs = ni.getInetAddresses();
+				while (inetAddrs.hasMoreElements()) {
+					InetAddress inetAddr = inetAddrs.nextElement();
+					String hostPort = inetAddr.getHostAddress() + ":" + proxyPort;
+
+					try {
+						HttpMessage newRequest = getNewMsg();
+						newRequest.getRequestHeader().setHeader("Proxy", hostPort);
+						try {
+							sendAndReceive(newRequest, false);
+						} catch (Exception e) {
+							// Ignore
+						}
+					
+						if (listener.isMsgReceived()) {
+							// the server is vulnerable
+							bingo(getRisk(), //Risk
+									Alert.CONFIDENCE_HIGH, //Confidence/Reliability
+									getName(), //Name
+									getDescription(), //Description
+									getBaseMsg().getRequestHeader().getURI().toString(), //Original URI
+									null, //Param
+									"Proxy: " + hostPort, //Attack
+									Constant.messages.getString(
+											MESSAGE_PREFIX + "otherinfo", listener.getMsgUrl()), //OtherInfo 
+									getSolution(), //Solution
+									"", //Evidence
+									getCweId(), //CWE ID
+									getWascId(), //WASC ID
```

Comment:
WASC-20 Improper Input Handling and CWE-441: Unintended Proxy or Intermediary ('Confused Deputy') ?